File: settlement/room/service/arena/RoomArenaWork.java
package settlement.room.service.arena;

import settlement.room.main.RoomInstance;
import snake2d.util.datatypes.COORDINATE;
import snake2d.util.datatypes.RECTANGLE;

public interface RoomArenaWork {

	public abstract COORDINATE gladiatorGetSpot(RoomInstance ins);
	public RECTANGLE gladiatorArea(int tx, int ty);
	public boolean gladiatorInArena(int tx, int ty);
	public void gladiatorDrawMakeSheer(COORDINATE coo);
	public RoomInstance reserveDeath(COORDINATE coo);
	public void unreserveDeath(int tx, int ty);
	
	public int executions();
	public int executionsMax();
	public int executions(RoomInstance ins);
	public int executionsMax(RoomInstance ins);
}

File: LegacySource/settlement/room/service/arena/RoomArenaWork.cs
using settlement.room.main;
using snake2d.util.datatypes;

namespace settlement.room.service.arena
{
    public interface RoomArenaWork
    {
        COORDINATE GladiatorsGetSpot(RoomInstance ins);
        RECTANGLE GladiatorsArea(int tx, int ty);
        bool GladiatorsInArena(int tx, int ty);
        void GladiatorsDrawMakeSheer(COORDINATE coo);
        RoomInstance ReserveDeath(COORDINATE coo);
        void UnreserveDeath(int tx, int ty);
        
        int Executions();
        int ExecutionsMax();
        int Executions(RoomInstance ins);
        int ExecutionsMax(RoomInstance ins);
    }
}